app/src/main/java/ch/epfl/sweng/SDP/home/leaderboard/Player.java

```diff
@@ -0,0 +1,127 @@
+     * @return 1 if this is larger, -1 if this is smaller, 0 else
+     */
+    @Override
+    public int compareTo(Object object) {
+        int compareTrophies = -this.trophies.compareTo(((Player) object).trophies);
+        if (compareTrophies == 0) {
+            return this.username.compareTo(((Player) object).username);
+        }
+        return compareTrophies;
+    }
+
+    boolean playerNameContainsString(String query) {
+        return username.toUpperCase().contains(query.toUpperCase());
+    }
+
+    /**
+     * Converts this player into a LinearLayout that will be displayed on the leaderboard.
+     *
+     * @param index   of the player
+     * @return LinearLayout that will be displayed
+     */
+    @SuppressLint("NewApi")
+    LinearLayout toLayout(int index) {
+        TextView usernameView = new TextView(context);
+        Resources res = context.getResources();
+        styleView(usernameView, rank + ". " + username, res.getColor(
+                isCurrentUser ? R.color.colorPrimaryDark : R.color.colorDrawYellow),
+                new LinearLayout.LayoutParams(0, LinearLayout.LayoutParams.WRAP_CONTENT, 4));
+        usernameView.setPadding(0, 10, 0, 10);
+
+        TextView trophiesView = new TextView(context);
+        styleView(trophiesView, trophies.toString(),
+                res.getColor(R.color.colorPrimaryDark),
+                new LinearLayout.LayoutParams(0,
+                        LinearLayout.LayoutParams.WRAP_CONTENT, 2));
+
+        trophiesView.setTextAlignment(RelativeLayout.TEXT_ALIGNMENT_TEXT_END);
+
+        ImageView leagueView = new ImageView(context);
+        leagueView.setLayoutParams(new LinearLayout.LayoutParams(0,
+                LinearLayout.LayoutParams.MATCH_PARENT, 1));
+        leagueView.setImageResource(getLeagueImageId(league));
+
+        final FriendsButton friendsButton =
+                new FriendsButton(context, this, index, isCurrentUser);
+
+        LinearLayout entry = addViews(new LinearLayout(context),
+                new View[]{usernameView, trophiesView, leagueView, friendsButton});
+
+        entry.setBackgroundColor(res.getColor(
+                isCurrentUser ? R.color.colorDrawYellow : R.color.colorLightGrey));
+        entry.setPadding(30, 10, 30, 10);
+
+        return entry;
+    }
+
+    private LinearLayout addViews(LinearLayout layout, View[] views) {
+        for (int i = 0; i < views.length; ++i) {
+            layout.addView(views[i]);
+        }
+        return layout;
+    }
+
+    private void styleView(TextView view, String text, int color,
+                           LinearLayout.LayoutParams layoutParams) {
+        view.setText(text);
+        view.setTextSize(20);
+        view.setMaxLines(1);
+        view.setTextColor(color);
+        Typeface typeMuro = Typeface.createFromAsset(context.getAssets(), "fonts/Muro.otf");
+        view.setTypeface(typeMuro);
+        view.setLayoutParams(layoutParams);
+    }
+
+    void setRank(int rank) {
```

Comment:
Is it used only for testing? We could add the VisibleForTesting annotation if it is the case

Reply:
This is not my function, but as far as I understood this is used for setting the rank before the username (e.g. 1. PICASSO, 2. GYRO, etc). So it must be package private.